### PR TITLE
Chore/update deps

### PR DESCRIPTION
**Pull request by TobTheRock**

### CODE REVIEW

File: examples/generate_headers.rs [hunk 1]
```diff
@@ -5,9 +5,13 @@ use sframe::{
...
     header::{Deserialization, Serialization},
 };
+use std::fmt::Write;
 
 fn bin2string(bin: &[u8]) -> String {
-    bin.iter().map(|x| format!("{x:08b} ")).collect()
+    bin.iter().fold(String::new(), |mut output, x| {
+        let _ = write!(output, "{x:08b} ");
+        output
+    })
```
Comment (hoodie) [inline examples/generate_headers.rs:14]: oh that lint

File: Cargo.toml [hunk 2]
```diff
@@ -24,32 +24,30 @@ log = "0.4"
 thiserror = "1.0"
 
 [dependencies.ring]
-version = "0.16"
+version = "0.17"
 optional = true
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]
-ring = { version = "0.16", features = ["wasm32_c"], optional = true }
+ring = { version = "0.17", features = ["wasm32_unknown_unknown_js"], optional = true }
```
Comment (hoodie) [inline Cargo.toml:31]: do we have an open source way to varify that this still works afterwards?

File: examples/generate_headers.rs (hunk 1 above)
Comment (TobTheRock) [reply examples/generate_headers.rs:14]: Not sure why those lints are not triggered in our weekly built, I have to double check that.

File: Cargo.toml (hunk 2 above)
Comment (TobTheRock) [reply Cargo.toml:31]: Unfortunately we can only verify that it still compiles :/ 
But I will double check with our closed source WASM implementation.

Comment (TobTheRock) [reply Cargo.toml:31]: Maybe we should add an example/test with nodejs for the future.

Comment (TobTheRock) [reply Cargo.toml:31]: Maybe we should add a minimal wasm example ?

Comment (TobTheRock) [reply Cargo.toml:31]: Works like a charm btw ;)

**Review by hoodie — APPROVED** [review]

**Pull request merged**